bayesian_exercises/golf_model_comparisons/golf.jl

```diff
@@ -0,0 +1,386 @@
+### A Pluto.jl notebook ###
+# v0.11.14
+
+using Markdown
+using InteractiveUtils
+
+# ╔═╡ 732bdf94-f6c8-11ea-35ae-490c5536996d
+begin
+	using Turing
+	using HTTP
+	using CSV
+	using Plots
+	using StatsFuns: logistic
+	using StatsPlots
+	using StatsBase
+end
+
+# ╔═╡ ce2eb27c-f6c8-11ea-02c8-d1dcefce145d
+begin
+	resp = HTTP.get("https://raw.githubusercontent.com/stan-dev/example-models/master/knitr/golf/golf_data.txt");
+
+	data = CSV.read(IOBuffer(resp.body), normalizenames = true, header = 3)
+	x, n, y = (data[:,1], data[:,2], data[:,3]);
+end
+
+# ╔═╡ 46428662-f6c9-11ea-3835-2dde7344ca40
+begin
+	pj = y ./ n
+	error = @. sqrt(pj * (1 - pj) / n)
+end
+
+# ╔═╡ 925fb2d6-f6c9-11ea-245f-f33635531f50
+scatter(x, pj, yerror=error, ylim=(0,1), xlabel="Distance from hole (ft)", ylabel="Probability of success", legend=false)
+
+# ╔═╡ 205dc84c-f6cb-11ea-3f31-c18a34f651b1
+begin
+	@model golf_logistic(x, y, n, J) = begin
+		# parameters
+		a ~ Normal(0, 1)
+		b ~ Normal(0, 1)
+		
+		for i in 1:J
+			p = logistic(a + b * x[i])
+			y[i] ~ Binomial(n[i], p)
+		end
+	end
+end
+
+# ╔═╡ a588cb4c-f6cc-11ea-31af-11c4e68303b1
+chn = sample(golf_logistic(x, y, n, length(x)), NUTS(), MCMCThreads(), 4000, 4);
```

Comment:
This line needs unpacking. First argument to `sample` is clearly the model, what are the rest?